### PR TITLE
Encode params

### PR DESCRIPTION
**Pull request by Mister-42**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  |

### CODE REVIEW

**Comment by Mister-42** [comment]
Decision making time:
* Use raurlencode for RFC 3986 compliance
* Use urlencode to be able to maintain the same URLs as in Yii 2

**Review by samdark — APPROVED** [review]

**Review by rustamwin — APPROVED** [review]

File: src/UrlMatcher.php [hunk 1]
```diff
@@ -60,6 +60,16 @@ final class UrlMatcher implements UrlMatcherInterface
...
 
+    /**
+     * Yii Framework encodes URLs differently than previous versions. If you are
+     * migrating a project from older versions, you can set this value to `true`
+     * to keep URLs encoded the same way.
+     * Default `false` is RFC 3986 compliant
+     *
+     * @var bool
+     */
+    private bool $yii2Compat = false;
```
Comment (samdark) [inline src/UrlMatcher.php:71]: I don't think we need it.

Comment (samdark) [reply src/UrlMatcher.php]: Then we need at least to name it differently.

Comment (Mister-42) [reply src/UrlMatcher.php]: This function is not for those people. Also, in params there is a clear explanation when to set it to true.

Comment (samdark) [reply src/UrlMatcher.php]: Well, I think we can name it better by what it does i.e. `encodeWithRFC` or `useUrlencode`, or `useRawurlencode` or something.

Comment (Mister-42) [reply src/UrlMatcher.php]: What it does is making router package compatible with how Yii 2 deals with parameters. Below that it indeed chooses between `urlencode` and `rawurlencode`, but in my opinion those names you suggested does not make anything clearer.

I you pick another name I will of course follow your lead, I just wanted to make you aware of my reasoning behind the name.

File: src/UrlGenerator.php [hunk 2]
```diff
@@ -20,6 +20,7 @@
 final class UrlGenerator implements UrlGeneratorInterface
 {
     private string $uriPrefix = '';
+    private bool $yii2Compat = false;
```
Comment (rustamwin) [inline src/UrlGenerator.php:23]:
```suggestion
    private bool $encodeRaw = true;
```

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks for handling it 👍

**Pull request merged**